### ISSUE TITLE
Compile and Qt Creator usage tutorial

### ISSUE THREAD

**Issue by Dogrtt** [libqgit2/libqgit2#10]
We need a small tutorial, how we can compile and use **libqgit2.**
What **CMake** flags we must use in configuration?
What version of **libgit2** we must use?
How can we connect **.dll** and **.lib** files to project?

**Comment by akikoskinen** [comment]
This doesn't sound like a bad proposition at all. Some documentation would definitely help newcomers start using the code. Hopefully someone will volunteer to write something up. I may be willing to write something up (not promising).

`.dll` and `.lib` sound like Windows to me and I know nothing about those, so somebody else has to write about those. Then again, they sound like libraries so isn't the short version something like "link your code against those libraries"?

**Comment by Dogrtt** [comment]
Is it right to use 0.27.5 version of libgit2 to build last libqgit2 library?
Also, I thought that I can write in this issue all about errors that I receive when I trying to use builded libraries, on linux mint and on Windows 10 x64. Is it acceptable?

**Comment by mattixiv** [comment]
Yes, `master` branch should work with 0.27.x. Libqgit2 doesn't have dependencies other than Qt5 and libgit2, so I suspect what you're actually having trouble with is building libgit2. Libgit2 relies on pkg-config to find some libraries that don't have an official CMake module, which makes things difficult for Windows developers.

But before I start guessing too much, I think it would be better to know what the actual problem is. IMO there shouldn't be any harm in reporting it in detail here.

**Comment by Johnyyyk** [comment]
This error is obtained by using the method of creating a repository (or other operations) in debug mode, but this error is not in the release. СMakeCache is attached. Mb we really made a mistake in cmake entries. Dynamic libraries are added to the project and are in the project folder.
![error in debug](https://user-images.githubusercontent.com/36297935/47500588-e16e4280-d86b-11e8-8d45-df2da477479b.jpg)
[CMakeCache.txt](https://github.com/libqgit2/libqgit2/files/2514896/CMakeCache.txt)

**Comment by mattixiv** [comment]
It's difficult to tell from just the cached values if something is wrong with the build. But that assertion would suggest that the problem isn't there.

That assertion is triggered by some error in memory management. For instance you might be deleting pointers that have not been allocated by `new`.

Whether the error is in the source code of `test_qgit.exe` or in libqgit2, libgit2 or Qt libraries I don't know.
